scripts/migration/schema-0.2-GPII-4014/createSimulatedGpiiKeys.js

```diff
@@ -0,0 +1,211 @@
+            "password": null,
+            "email": null,
+            "preferences": prefs,
+            "timestampCreated": currentTime,
+            "timestampUpdated": null
+        };
+        newDocs.push(newGpiiKey);
+        newDocs.push(newPrefsSafe);
+    };
+    return newDocs;
+};
+
+/**
+ * Log how many GPII documents were updated.
+ * @param {String} responseString - Response from the database (ignored)
+ * @param {Object} options - Object containing the set of documents:
+ * @param {Array} options.numOfKeysToCreate - Total number of GPII keys to create.
+ * @param {Array} options.numOfCreatedKeys - Total number of GPII keys that have been created.
+ * @return {Number} - the number of documents created.
+ */
+gpii.migration.GPII4014.logUpdateDB = function (responseString, options) {
+    var togo = fluid.promise();
+    options.numOfCreatedKeys = Number(options.numOfCreatedKeys) + Number(options.numToCreateInThisBatch);
+    if (Number(options.numOfCreatedKeys) < Number(options.numOfKeysToCreate)) {
+        togo.resolve("Created " + options.numOfCreatedKeys + " of requested " + options.numOfKeysToCreate + " GPII keys.");
+    } else {
+        togo.reject({
+            errorCode: "GPII-CREATED-ENOUGH",
+            message: options.numOfCreatedKeys + " have been created."
+        });
+    }
+    return togo;
+};
+
+/**
+ * Configure one batch of creation.
+ * @param {Object} options - Object containing the set of documents:
+ * @param {Array} options.newDocs - The documents to create.
+ * @param {Array} options.numOfKeysToCreate - Total number of GPII keys to create.
+ * @param {Array} options.numOfCreatedKeys - Total number of GPII keys that have been created.
+ * @return {Promise} - The promise that resolves the creation of this batch.
+ */
+gpii.migration.GPII4014.createOneBatch = function (options) {
+    var numOfNeeded = options.numOfKeysToCreate - options.numOfCreatedKeys;
+    options.numToCreateInThisBatch = numOfNeeded > options.maxDocsInBatchPerRequest ? options.maxDocsInBatchPerRequest : numOfNeeded;
+
+    options.newDocs = gpii.migration.GPII4014.generateKeyData(options.numToCreateInThisBatch);
+
+    var details = {
+        dataToPost: options.newDocs,
+        responseDataHandler: gpii.migration.GPII4014.logUpdateDB
+    };
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Create GPII keys recursively.
+ * @param {Object} options - Object containing the set of information for GPII key creation.
+ */
+gpii.migration.GPII4014.createRecursive = function (options) {
+    var createPromise = gpii.migration.GPII4014.createOneBatch(options);
+
+    createPromise.then(
+        function (message) {
+            console.log(message);
+            gpii.migration.GPII4014.createRecursive(options);
+        },
+        function (error) {
+            if (error.errorCode === "GPII-CREATED-ENOUGH") {
+                console.log("Done: " + error.message);
+                process.exit(0);
```

Comment:
Rather fierce - better to return a promise and push the responsibility for exiting the process onto the caller